server/src/main/java/org/opensearch/action/search/AbstractSearchAsyncAction.java

```diff
@@ -171,6 +181,119 @@ abstract class AbstractSearchAsyncAction<Result extends SearchPhaseResult> exten
         this.indexRoutings = indexRoutings;
         this.results = resultConsumer;
         this.clusters = clusters;
+        if (searchListenersList != null && !(searchListenersList.isEmpty())) {
+            this.searchListenersList = searchListenersList;
+            this.searchRequestOperationsListener = new SearchRequestOperationsListener.CompositeListener(this.searchListenersList, logger);
```

Comment:
Do we need both of them as class members?

Reply:
Yes. We certainly need searchRequestOperationsListener.
We could have avoided searchListenersList though by exposing a getter of it in listener itself.
 This current logic looks okay to me, let me know if this is a concern to you.